src/codecs/decoders.ts

```diff
@@ -0,0 +1,42 @@
+import * as browserJPEG from './browser-jpeg/decoder';
+import * as browserPNG from './browser-png/decoder';
+import * as browserWebP from './browser-webp/decoder';
+import * as wasmWebP from './webp/decoder';
+
+export interface Decoder {
+  name: string;
+  decode(file: File): Promise<ImageBitmap>;
+  isSupported(): Promise<boolean>;
+  supportedMimeTypes: string[];
+  supportedExtensions: string[];
+}
+
+// We load all decoders and filter out the unsupported ones.
+export const decodersPromise: Promise<Decoder[]> = Promise.all(
+  [
+    browserPNG,
+    browserJPEG,
+    wasmWebP,
```

Comment:
This is just so the wasm decoder takes precedence of the native one for testing purposes. Will remove before merging.

Reply:
Or rather: will move it back down before merging.